clients/render/components/Fields/FileInput.js

```diff
@@ -2,35 +2,50 @@ import {CalderaFormsFieldGroup, Fragment} from "../CalderaFormsFieldGroup";
 import {CalderaFormsFieldPropType} from "../CalderaFormsFieldRender";
 import PropTypes from 'prop-types';
 import Dropzone from 'react-dropzone';
+import { setFormInState } from '../../../state/actions/mutations'
 const CryptoJS = require("crypto-js");
 
-export const FileInput = (props) => {
-	const{shouldDisable,accept,field,describedById,onChange,style,className,multiple,multiUploadText,inputProps} = props;
-	const {
-		outterIdAttr,
-		fieldId,
-		fieldLabel,
-		fieldCaption,
-		required,
-		fieldPlaceHolder,
-		fieldDefault,
-		fieldValue,
-		fieldIdAttr,
-		isInvalid
-	} = field;
-
-	const onDrop = (acceptedFiles, rejectedFiles) => {
-		acceptedFiles.forEach(file => {
-			onChange(file);
+export class FileInput extends React.Component {
 
-		});
-	};
+  constructor(props) {
```

Comment:
@New0 Why use a stateful component here?

The disadvantage that I see is that the File object is now stored in two places -- this component's state and the form render component's state.

I'm not sure what the advantage is. I see that you are using `this.state.files` to build the preview, I do not see an advantage of this versus itterating on `props.fieldValue` which should be one File object or an array of File objects.